hamilton/caching/stores/sqlite.py

```diff
@@ -118,7 +122,15 @@ def set(
         # if the caller of ``.set()`` directly provides the ``node_name`` and ``code_version``,
         # we can skip the decoding step.
         if (node_name is None) or (code_version is None):
-            decoded_key = decode_key(cache_key)
+            try:
+                decoded_key = decode_key(cache_key)
+            except BaseException as e:
+                raise ValueError(
+                    f"Failed decoding the cache_key: {cache_key}.\n",
+                    "Was it manually created? Do `code_version` and `data_version` found in ",
```

Comment:
What's the error case that this corresponds to? Is a user ever going to hit this? If so, point to docs on how to use it (it's a bit unclear from the error message)

Reply:
This would happen if the user is meeting these conditions:
- using `SQLiteMetadataStore`
- passes a manually crafted key that isn't the result of `hamilton.caching.cache_key.create_key()` (i.e., it can't be parsed)
- they are not passing explicitly a `node_name` AND `code_version` so the metadata store needs to derive it.

e.g.,
```python
sqlite_metadata_store.set(
   cache_key="foo"   # not a properly formatted key
   data_version="any-string"  # this can be anything
)
```

This will never happen when using the adapter, only if the user is manually interacting with the metadata store.

## Why this odd behavior?
Metadata stores `.set()` only required kwargs `cache_key` and `data_version`. This provides limited information for cache management. Storing the `node_name` and `code_version` in the metadata store allows to delete entries by node name or version (not implemented yet). On the other, forcing this information to be available can add constraints

Reply:
For later, I think it could be a `__init__()` kwarg for `MetadataStore` to toggle indexing `node_name` and `code_version`. This mostly affects SQL-based metadata store, but `node_name` indexing can also be possible in key-value stores via tags

Reply:
Yep, so add comments/more to the error message to clarify

Reply:
yup, I improved it in the latest code version